inginious-autotest

```diff
@@ -0,0 +1,255 @@
+    :param output1: output from the yaml
+    :param output2: output from the new job
+    :param keys: list of string, representing the keys for each value in the output
+    :return: a dict, whose the form is
+        {k: t,...} where k is a key whose the associated values in the outputs are different and t is a tuple containing
+        the values from both the outputs
+    """
+    return {
+            keys[i]: (output1[i], output2[i]) for i in range(len(keys))
+            if compare_output(output1[i], output2[i], keys[i]) and output1[i] is not None and output2[i] is not None
+        }
+
+
+def compare_output(output1, output2, key):
+    """
+    Compare atomic elements from the outputs
+    :param output1: atomic data structure from the yaml
+    :param output2: atomic data structure from the new job
+    :param key: key associated
+    :return: True if output1 and output2 are different, False otherwise
+    """
+
+    def result_compare(output1, output2):
+        """
+        Compare for the result key
+        :param output1: string
+        :param output2: tuple (string 1, string 2)
+        :return: True if output1 is different from the string 1 for output2
+        """
+        return output1 != output2[0]
+
+    def problems_compare(output1, output2):
+        """
+        Compare for the problems key
+        :param output1: dict whose each value is a list
+        :param output2: dict whose each value is a tuple
+        :return: True if output1 != output2, False otherwise
+        """
+        for k in output1:  # same keys in output1 and output2
+            l = output1[k]  # in output1, values are lists of size 2
+            t = output2[k]  # in output2, values are tuples of size 2
+            if l[0] != t[0] or l[1] != t[1]:
+                return True
+        return False
+
+    def archive_compare(output1, output2):
+        """
+        Return False because archives are always different
+        """
+        return False
+
+    def generic_compare(output1, output2):
+        """
+        Generic compare for keys
+        :param output1: string
+        :param output2: string
+        :return: see code
+        """
+        return output1 != output2
+
+    func = {
+        "result": result_compare,
+        "problems": problems_compare,
+        "archive": archive_compare
+    }
+    return func.get(key, generic_compare)(output1, output2)
+
+
+def test_task(yaml_data, task, client, client_sync):
+    """
+    Test the task by comparing the new outputs with the old ones
+    :param yaml_data: dict corresponding to the yaml output file for the task
+    :param task: Task object corresponding to the yaml data
+    :param client: backend client of type Client
+    :param client_sync: ClientSync object
+    :return: dict whose the format is specified in compare_all_outputs function doc
+    """
+    if task.get_environment() not in client.get_available_containers():
+        time.sleep(10)
```

Comment:
Let's try with 1 rather than 10 ;-)

Reply:
Done and it's working locally and on Travis